scripts/machine-launch-script.sh

```diff
@@ -0,0 +1,305 @@
+    #    for notes on using the conda compilers
+    # elfutils has trouble building with gcc 11 for something that looks like it needs to be fixed upstream
+    # ebl_syscall_abi.c:37:64: error: argument 5 of type 'int *' declared as a pointer [-Werror=array-parameter=]
+    #    37 | ebl_syscall_abi (Ebl *ebl, int *sp, int *pc, int *callno, int *args)
+    #       |                                                           ~~~~~^~~~
+    # In file included from ./../libasm/libasm.h:35,
+    #                  from ./libeblP.h:33,
+    #                  from ebl_syscall_abi.c:33:
+    # ./libebl.h:254:46: note: previously declared as an array 'int[6]'
+    #   254 |                             int *callno, int args[6]);
+    #       |                                          ~~~~^~~~~~~
+    #
+    # pin to gcc=10 until we get that fixed.
+
+    CONDA_PACKAGE_SPECS+=( gcc=10 gxx=10 binutils conda-gcc-specs )
+
+    # if building riscv-toolchain from source, we need to use bison=3.4 until we have
+    # https://github.com/riscv-collab/riscv-binutils-gdb/commit/314ec7aeeb1b2e68f0d8fb9990f2335f475a6e33
+    CONDA_PACKAGE_SPECS+=( bison=3.4 )
+
+    # poky deps
+    CONDA_PACKAGE_SPECS+=( python=3.8 patch texinfo subversion chrpath git wget )
+    # qemu deps
+    CONDA_PACKAGE_SPECS+=( gtk3 glib pkg-config bison flex )
+    # qemu also requires being built against kernel-headers >= 2.6.38 because of it's use of
+    # MADV_NOHUGEPAGE in a call to madvise.  See:
+    # https://man7.org/linux/man-pages/man2/madvise.2.html
+    # https://conda-forge.org/docs/maintainer/knowledge_base.html#using-centos-7
+    # obvi this would need to be made linux-specific if we supported other MacOS or Windows
+    CONDA_PACKAGE_SPECS+=( "kernel-headers_linux-64>=2.6.38" )
+    # firemarshal deps
+    CONDA_PACKAGE_SPECS+=( rsync psutil doit gitpython humanfriendly e2fsprogs ctags bison flex expat )
+    # cross-compile glibc 2.28+ deps
+    # current version of buildroot won't build with make 4.3 https://github.com/firesim/FireMarshal/issues/236
+    CONDA_PACKAGE_SPECS+=( make!=4.3 )
+    # build-libelf wants autoconf
+    CONDA_PACKAGE_SPECS+=( autoconf automake libtool )
+    # other misc deps
+    CONDA_PACKAGE_SPECS+=(
+        sbt \
+        ca-certificates \
+        mosh \
+        gmp \
+        mpfr \
+        mpc \
+        zlib \
+        vim \
+        git  \
+        openjdk \
+        gengetopt \
+        libffi \
+        expat \
+        libusb1 \
+        ncurses \
+        cmake \
+        graphviz \
+        expect \
+        dtc \
+        verilator==4.034 \
+    )
+
+    if [[ "$CONDA_ENV_NAME" == "base" ]]; then
+        # NOTE: arg parsing disallows installing to base but this logic is correct if we ever change
+        CONDA_SUBCOMMAND=install
+        CONDA_ENV_BIN="${CONDA_INSTALL_PREFIX}/bin"
+    else
+        CONDA_ENV_BIN="${CONDA_INSTALL_PREFIX}/envs/${CONDA_ENV_NAME}/bin"
```

Comment:
Is it standard practice to put the envs in `/opt/conda`? This would require sudo? Why don't we just install it to `chipyard/conda-env` (for example)

Reply:
I think we can have it default to a Chipyard path, but we want to give the user the option to override it to put it in a global area.